pkg/operator/options.go

```diff
@@ -0,0 +1,94 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"runtime/debug"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/util/flowcontrol"
+	"k8s.io/utils/clock"
+	"knative.dev/pkg/configmap/informer"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/aws/karpenter/pkg/apis"
+	"github.com/aws/karpenter/pkg/events"
+	"github.com/aws/karpenter/pkg/utils/injection"
+	"github.com/aws/karpenter/pkg/utils/options"
+	"github.com/aws/karpenter/pkg/utils/project"
+)
+
+const (
+	appName   = "karpenter"
+	component = "controller"
+)
+
+var (
+	scheme = runtime.NewScheme()
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(apis.AddToScheme(scheme))
+}
+
+func NewOptionsWithManagerOrDie() (Options, manager.Manager) {
+	opts := options.New().MustParse()
```

Comment:
Is there a way for us to collapse these two options concepts? I'm not sure what the separation of concerns is, and they feel like the same thing, e.g. a global bag of things that anyone is allowed to use. Perhaps just hoist /utils/options up here?

Reply:
I think it's a good point, but probably something that I want to handle in a separate PR, primarily because the current `options` are injected into context and there needs to be more of a discussion around what we do/don't want to inject into context.

Reply:
Also, this PR already touches quite a bit so I'd be more apt to have a follow-up as another cleanup tassk